docs/DeveloperGuide.md

```diff
@@ -532,133 +535,213 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+1. User requests to edit a friend and specifies a `FRIEND_ID` and new `NAME` to update.  
+2. gitGud updates friend and informs user of successful edit of friend.  
 
-    Use case ends.
+   Use case ends.  
 
 **Extensions**
 
-* 2a. The given `FRIEND_ID` is invalid.
+* 1a. gitGud detects that the `FRIEND_ID` does not exist.  
+    * 1a1. gitGud informs user that it has no friend was found with specified `FRIEND_ID`.  
+    * 1a2. User makes another request to edit a friend with a different `FRIEND_ID` for an existing friend.  
+    * Use case continues from step 2.  
+
+* 1b. gitGud detects that the `NAME` provided is invalid and does not meet specifications provided in User Guide.
+    * 1b1. gitGud informs user that `NAME` provided is invalid.
+    * 1b2. User makes another request to edit friend with `FRIEND_ID` with a different valid `NAME`.
+    * Use case continues from step 2.
+
+**Use case: UC03 - Delete a friend**
+
+**MSS**
+
+1. User requests to delete a friend using a friend's `FRIEND_ID`.  
+2. gitGud deletes the friend from friend list and informs user of successful deletion of friend.  
 
-    * 2a1. gitGud shows an error message.
+    Use case ends.
 
-      Use case resumes at step 1.
+**Extensions**
 
+* 1a. gitGud detects that the `FRIEND_ID` does not exist.  
+    * 1a1. gitGud informs user that it has no friend was found with specified `FRIEND_ID`.  
+    * 1a2. User makes another request to delete a friend with a different `FRIEND_ID` for an existing friend.
+    * Use case continues from step 2.  
 
-**Use case: UC03 - List friends whose `FRIEND_ID` contains a keyword**
+**Use case: UC04 - Link a game to a friend**
 
 **MSS**
 
-1. User requests to list friends in gitGud using a keyword filter.
-2. gitGud shows a filtered list of friends whose `FRIEND_ID` contains the keyword.
+1. User requests to link a friend using a friend's `FRIEND_ID`, a game's `GAME_ID` and an `IN_GAME_USERNAME`.
+2. gitGud links the friend to the game, stores their in-game username and informs user of successful linking of friend.
 
    Use case ends.
 
 **Extensions**
 
-* 2a. The user does not provide a keyword.
+* 1a. gitGud detects that the `FRIEND_ID`/`GAME_ID` does not exist.
+    * 1a1. gitGud informs user that it has no friend/game was found with specified `FRIEND_ID`/`GAME_ID`.  
+    * 1a2. User makes another request to link a friend with a game, with a different `FRIEND_ID`/`GAME_ID` for an existing friend/game.  
+    * Use case continues from step 2.     
+
+* 1b. gitGud detects that the `IN_GAME_USERNAME` provided is invalid and does not meet specifications provided in User Guide.
+    * 1b1. gitGud informs user that `IN_GAME_USERNAME` provided is invalid.
+    * 1b2. User makes another request to link a friend with a game, with existing `FRIEND_ID` and `GAME_ID` and valid `IN_GAME_USERNAME`.
+    * Use case continues from step 2.
 
-    * 2a1. All friends are listed.
+**Use case: UC05 - Unlink a game from a friend**
```

Comment:
```suggestion
**Use case: UC05 - Unlink a friend from a game**
```

Reply:
Same as above,  I am trying to follow the UG's